pkg/service/service.go

```diff
@@ -1344,8 +1344,25 @@ func (s *Service) createSVCInfoIfNotExist(p *lb.SVC) (*svcInfo, bool, bool,
 	prevSessionAffinity := false
 	prevLoadBalancerSourceRanges := []*cidr.CIDR{}
 
+	proto := p.Frontend.L3n4Addr.L4Addr.Protocol
```

Comment:
this deserves a comment explaining the logic here and is not clear to me without having to understand the whole code base:
1. we store the original proto
2. we mutate everything to NONE
3. if does not exist with NONE with try with the original proto

did I get it right?

Reply:
that's correct :+1: 

> this deserves a comment explaining the logic here

yes, I'm planning to sprinkle some extra comments once I start cleaning up all this WIP commits